File: listings/probabilistic-graph.kt
// Kotlin
val clausesParser = ClausesParser.withOperators(PROBLOG_OPERATORS)
val probabilisticTheory = clausesParser.parseTheory("(*@\color{magenta}$\langle$@*)theory from (*@\color{magenta}\cref{fig:valid-probabilistic-graph}$\rangle$@*)")
val problogSolver = Solver.problog.solverWithDefaultBuiltins(staticKb = probabilisticTheory)
val goal = Struct.of("path", Var.of("From"), Var.of("To"))
for (solution in problogSolver.solve(goal, SolveOptions.allLazily().probabilistic()))
    if (solution.isYes)
        println("yes: ${solution.solvedQuery} with probability ${solution.probability}")

File: listings/probabilistic-graph.py
# Python
probabilisticTheory = parse_theory("(*@\color{magenta}$\langle$@*)theory from (*@\color{magenta}\cref{fig:valid-probabilistic-graph}$\rangle$@*)", PROBLOG_OPERATORS)
problogSolver = problog_solver(static_kb=probabilisticTheory)
query = struct('path', var('From'), var('To'))
for solution in problogSolver.solve(query, solve_options(lazy=True, probabilistic=True)):
    if solution.is_yes:
        print(f"yes: {solution.solved_query} with probability {probability(solution)}")
